gateway/crates/gateway-binary/src/args.rs

```diff
@@ -1,13 +1,32 @@
-#[cfg(feature = "lambda")]
 mod lambda;
 mod log;
-#[cfg(not(feature = "lambda"))]
 mod std;
 
+use ::std::net::SocketAddr;
+
+use clap::Parser;
+use federated_server::{Config, GraphFetchMethod};
+use grafbase_tracing::otel::layer::BoxedLayer;
 pub(crate) use log::LogLevel;
+use tracing::Subscriber;
+use tracing_subscriber::registry::LookupSpan;
 
-#[cfg(feature = "lambda")]
-pub(crate) use lambda::Args;
+pub(crate) trait Args {
+    fn listen_address(&self) -> Option<SocketAddr>;
+    fn log_level(&self) -> Option<LogLevel>;
+    fn fetch_method(&self) -> anyhow::Result<GraphFetchMethod>;
+    fn config(&self) -> anyhow::Result<Config>;
+    fn log_format<S>(&self) -> BoxedLayer<S>
+    where
+        S: Subscriber + for<'span> LookupSpan<'span> + Send + Sync;
+}
 
-#[cfg(not(feature = "lambda"))]
-pub(crate) use self::std::Args;
+pub(crate) fn parse() -> impl Args {
```

Comment:
:3

Reply:
didn't want to write anymore code hidden behind a `not(feature = "lambda")` flag 😅